frontend/components/transaction/TransactionFeesTable.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+    <div v-if="isWalletConnected" class="flex flex-col space-y-1 text-gray-700 dark:text-gray-100">
+        <div v-if="swapTransactionFeeETH" class="flex justify-between">
+            <div>Swap Transaction Fee</div>
+            <div>
+                <FormatCurrency :value="swapTransactionFeeETH" :decimals="5" :currency="ETH" />
+            </div>
+        </div>
+        <div v-else-if="bidTransactionFeeETH" class="flex justify-between">
+            <div>Bid Transaction Fee</div>
+            <div>
+                <FormatCurrency :value="bidTransactionFeeETH" :decimals="5" :currency="ETH" />
+            </div>
+        </div>
+        <div class="flex justify-between">
+            <div>Wallet Authorization Fee <Icon v-if="isWalletAuthed" type="check" class="text-green-500" /></div>
+            <div>
+                <FormatCurrency :value="authTransactionFeeETH" :decimals="5" :currency="ETH" />
+            </div>
+        </div>
+        <div class="flex justify-between">
+            <div>
+                Collateral Authorization Fee <Icon v-if="isCollateralAuthed" type="check" class="text-green-500" />
+            </div>
+            <div>
+                <FormatCurrency :value="authTransactionFeeETH" :decimals="5" :currency="ETH" />
+            </div>
+        </div>
+        <hr />
+        <div class="flex justify-between font-bold">
+            <div>Total</div>
+            <div>
+                <FormatCurrency
+                    v-if="combinedSwapFeesETH"
+                    :value="combinedSwapFeesETH"
+                    :decimals="5"
+                    :currency="ETH"
+                />
+                <FormatCurrency
+                    v-else-if="combinedBidFeesETH"
+                    :value="combinedBidFeesETH"
+                    :decimals="5"
+                    :currency="ETH"
+                />
+            </div>
+        </div>
+    </div>
+    <div v-else>Connect a wallet for an approximate fee breakdown</div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Icon } from 'ant-design-vue';
+import BigNumber from 'bignumber.js';
+import FormatCurrency from '~/components/utils/FormatCurrency.vue';
+
+export default Vue.extend({
+    components: {
+        FormatCurrency,
+        Icon,
+    },
+    props: {
+        swapTransactionFeeETH: {
```

Comment:
I have an alternative idea, to make this component easier to extend.

Why don't you pass an object of fees: 

```ts
fees = {
   'Swap Fee': new Bignumber(5),
   ... etc etc
}
``` 

Afterwards you compute the total amount as a computed property. This allows us more flexibility when adding/removing fees. (You create the fee object in the parent component where we receive an auction/surplus auction object)

More insight:
This allows us the remove a lot of these props. The parent component then checks if the wallet is auth for example and only hands down the needed fees.